public/src/components/teacher/Results.tsx

```diff
@@ -143,6 +143,23 @@ export class Results extends React.Component<IResultsProps, IResultsState> {
             const ans = await this.props.onSubmissionStatusUpdate(selected);
             if (ans) {
                 selected.approvedDate = new Date().toLocaleString();
+
+                if (current && selected.groupid > 0) {
+                    this.state.allSubmissions.forEach((e) => {
+                        if (e.enrollment.getGroup()?.getId() === selected.groupid) {
+                            e.labs.forEach((l) => {
+                                if (l.assignment.getId() === current.assignment.getId()) {
+                                    const double = l.submission;
```

Comment:
What does `double` mean?

Reply:
Yeah, not the best name choice on my side. In general, as this is the main result page, all submissions are grouped by students. As a result, submissions for group assignments will have copies of that submission for each group member. Here we update all the copies of a submission with the changed state locally to avoid re-fetching the whole course from the server.
Changed name to `currentSubmissionCopy`, what do you think?